pkg/apis/v2v/v1alpha1/resourcemapping_types.go

```diff
@@ -0,0 +1,83 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// ResourceMappingSpec defines the desired state of ResourceMapping
+// +k8s:openapi-gen=true
+type ResourceMappingSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "operator-sdk generate k8s" to regenerate code after modifying this file
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// +optional
+	OvirtMappings *OvirtMappings `json:"ovirt,omitempty"`
+}
+
+// OvirtMappings defines the mappings of ovirt resources to kubevirt
+// +k8s:openapi-gen=true
+type OvirtMappings struct {
```

Comment:
Can we have a more generic type for the mappings? Will mapping for VMware be different?
Having `type Mappings`, not bound to any provider we can implement the provider "plug-in" API in a more generic way in some of its parts - for example in validation, where the mapping needs to be verified we can pass just the `Mapping` instead of its properties separately.

Reply:
OK, sounds good :+1: